### PR TITLE
Update docs for dns and technical overview (closes #7 and #8)

### LINKED ISSUE #7: More Complete Explanation of how Dist Press publishes
@jackyzha0 I would love a step-by-step explanation of what happens when I publish to Dist press! I know you listed the protocols you work with, but it would be nice to understand the workflow, e.g. zip/ bundle files, use x API to publish to x protocol, use y api to publish to y protocol .... whatever else may happen!

### PR DESCRIPTION
**Pull request by jackyzha0**
@RangerMauve can you add the docs for BitTorrent when you have the time?

@YurkoWasHere wanted to ask for your review just to check the technical accuracy here. Some parts I'm not super sure about (esp some of the DNSLink magic)

### CODE REVIEW

**Comment by jackyzha0** [comment]
@walkerlj0 also take a look at this when you get the chance

File: pages/overview.mdx [hunk 1]
```diff
@@ -1,29 +1,111 @@
...
+  "pubKey": string,
+  "dnslink": string,
+  "enabled": boolean,
+  "link": string 
+}
+```
 
+### Holepunch
+The associated logic for Holepunch is [in this file](https://github.com/hyphacoop/api.distributed.press/blob/main/protocols/hyper.ts).
+Holepunch similarly uses [`localdrive`](https://www.npmjs.com/package/localdrive) which overs a mutable file system API that is similar to Hyperdrive.
```
Comment (walkerlj0) [inline pages/overview.mdx:33]:
```suggestion
Holepunch similarly uses [`localdrive`](https://www.npmjs.com/package/localdrive) which offers a mutable file system API that is similar to Hyperdrive.
```

File: pages/overview.mdx [hunk 2]
```diff
@@ -1,29 +1,111 @@
...
+instance that is spun-up when Distributed Press starts up. This connection is managed by `js-ipfsd-ctl`.
 
-Clients make authenticated HTTP requests to make updates to resources 
-like publishers, admins, and sites.
+We then copy the files to an IPFS [Mutable File System (MFS)](https://docs.ipfs.tech/concepts/file-systems/) and 
+publish it to IPFS. When finished, this returns us a CID. We update the IPNS entry for the site ID to this resulting CID.
 
-The API server is responsible for locally storing site data to serve and 
-seed over the various [peer-to-peer protocols](./protocols) that Distributed Press supports.
+This CID and accompanying IPNS entry is used to populate the `links` section:
```
Comment (walkerlj0) [inline pages/overview.mdx:18]: The links section of what? Is there an example / reference config file you can point to?

File: pages/overview.mdx [hunk 3]
```diff
@@ -1,29 +1,111 @@
...
-## DNS
-The DNS server runs alongside the Distributed Press API and is responsible
-for resolving DNS queries for sites that it hosts.
+We create a drive for each site and then copy the files into it. This is done through `localdrive` as it has a feature to
+mirror files from your filesystem to the drive.
 
-This means that if you own the domain for the site you host (e.g. `distributed.press`),
-you can add an `NS` record through your domain registrar to tell it to look at
-Distributed Press for where to find where the site is actually hosted.
+When finished, we get the drive URL and similarly populate the `links` section:
```
Comment (walkerlj0) [inline pages/overview.mdx:38]: Where can once access the HTTP links and CIDs that one has published to? 

For example, with Github Actions, this is output in the Actions workflow. Not sure how Sutty shares this, but would be good to reference where one can find this

Comment (walkerlj0) [reply pages/overview.mdx:38]: dpress-actions-output

**Review by walkerlj0 — CHANGES_REQUESTED** [review]
Super helpful! I'm not sure myself about the dnslink stuff, but examples for the NS record would help too. Also one thing I noticed was that there was no reference for someone who has published to find the holepuch/ IPFS CIDs and HTTP links

File: pages/deployment/custom-domain.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+means that even if the Distributed Press instance you use goes down for whatever reason, you can *still* access your 
+content if at least one other person has a copy of the site and is hosting it.
+</Callout>
+
+## Custom Domain
+If you'd like your Distributed Site to be available via HTTP at your own custom domain (e.g. like how this site is available over `docs.distributed.press`),
+you can do that in one of two ways:
+
+### Using a Gateway
+If you'd like to serve content from a particular gateway, you can create a `CNAME` record with your domain registrar that points at the subdomain of the gateway.
```
Comment (RangerMauve) [inline pages/deployment/custom-domain.mdx:28]: Do we have an example of this working? I thought the subdomain hostname wasn't getting passed to the gateway when you CNAME.

Right now we CNAME to `api.distributed.press` for most things and have that resolve through NGINX and the distributed_press_served_sites list.

File: pages/overview.mdx [hunk 5]
```diff
@@ -1,29 +1,111 @@
 # Technical Overview
+## What happens when you publish a site to Distributed Press?
+When you [upload content to a site](./api-reference/site#uploading-content-to-a-site) on Distributed Press, the server first 
+untars the files to a temporary folder on the server.
```
Comment (RangerMauve) [inline pages/overview.mdx:4]:
```suggestion
extracts the files to a temporary folder on the server.
```

File: pages/overview.mdx (hunk 2 above)
Comment (RangerMauve) [reply pages/overview.mdx:18]: THe `links` section when you do a `GET /v1/sites/{your site domain/`.

E.g. this is what my blog looks like:

```
{
  "id": "blog.mauve.moe",
  "domain": "blog.mauve.moe",
  "protocols": {
    "http": true,
    "hyper": true,
    "ipfs": true,
    "bittorrent": true
  },
  "links": {
    "http": {
      "enabled": true,
      "link": "example-link"
    },
    "hyper": {
      "gateway": "https://blog-mauve-moe.hyper.hypha.coop/",
      "raw": "hyper://1m51x54k3fwbuec5z4edbathiq3aj6bew8b556ezrszizskogo3o/",
      "dnslink": "/hyper/1m51x54k3fwbuec5z4edbathiq3aj6bew8b556ezrszizskogo3o/",
      "enabled": true,
      "link": "hyper://blog.mauve.moe"
    },
    "ipfs": {
      "gateway": "https://blog-mauve-moe.ipns.ipfs.hypha.coop",
      "cid": "bafybeidpngig3g4gqrbooyrzxe4vl5jusad7om2chtl7vgrzcqbpilni5e",
      "pubKey": "ipns://k51qzi5uqu5dgxx5s83cvbehacxojuztwxnhnmxm1npizgve9e3fehpk2lqnf9/",
      "dnslink": "/ipns/k51qzi5uqu5dgxx5s83cvbehacxojuztwxnhnmxm1npizgve9e3fehpk2lqnf9/",
      "enabled": true,
      "link": "ipns://blog.mauve.moe/"
    },
    "bittorrent": {
      "gateway": "https://blog-mauve-moe.bt.hypha.coop/",
      "magnet": "magnet:?xt:urn:btih:bbad2f8321989a8395a6b1bac6240877ab86727a&xs=urn:btpk:416ce4db34ae1260d81c2aed16bd1eab3481e4e48f3c781aa9c057b15d900b71",
      "infoHash": "bittorrent://bbad2f8321989a8395a6b1bac6240877ab86727a/",
      "pubKey": "bittorrent://416ce4db34ae1260d81c2aed16bd1eab3481e4e48f3c781aa9c057b15d900b71/",
      "dnslink": "/bittorrent/416ce4db34ae1260d81c2aed16bd1eab3481e4e48f3c781aa9c057b15d900b71",
      "enabled": true,
      "link": "bittorent://blog.mauve.moe/"
    }
  },
  "public": true
}
```

**Comment by RangerMauve** [comment]
I can add some docs on BT next week. Still fixing some bugs with loading stuff (I think firewall issues?)

File: pages/deployment/custom-domain.mdx (hunk 4 above)
Comment (jackyzha0) [reply pages/deployment/custom-domain.mdx:28]: Gateways should still be able to resolve it though? Check the IPNS/DNSLink section of the gateway services: https://docs.ipfs.tech/concepts/ipfs-gateway/#gateway-services

Comment (RangerMauve) [reply pages/deployment/custom-domain.mdx:28]: I'm just not sure if CNAME would work. As I understand CNAME to a subdomain will get you the IP of the subdomain but the browser will still send the HOSTNAME of the original domain.

So even if `docs.distributed.press` pointed at `my-domain.hyper.hypha.coop`, the server at `hyper.coop` would see the request coming to `docs.distributed.press` and see that it's not a subdomain of `hyper.hypha.coop` and error out. (likely an HTTPS error due to certificate mismatch).

File: pages/overview.mdx [hunk 6]
```diff
@@ -1,29 +1,111 @@
...
+instance that is spun-up when Distributed Press starts up. This connection is managed by `js-ipfsd-ctl`.
 
-Clients make authenticated HTTP requests to make updates to resources 
-like publishers, admins, and sites.
+We then copy the files to an IPFS [Mutable File System (MFS)](https://docs.ipfs.tech/concepts/file-systems/) and 
+publish it to IPFS. When finished, this returns us a CID. We update the IPNS entry for the site ID to this resulting CID.
 
-The API server is responsible for locally storing site data to serve and 
-seed over the various [peer-to-peer protocols](./protocols) that Distributed Press supports.
+This CID and accompanying IPNS entry is used to populate the `links` section:
```
Comment (walkerlj0) [inline pages/overview.mdx:18]:
```suggestion
This CID and accompanying IPNS entry is used to populate [the `links` section](https://docs.distributed.press/api-reference/site#getting-the-information-of-site):
```

**Pull request merged**